libs/api/domains/rsk-company-info/src/lib/graphql/rsk-company-info.service.ts

```diff
@@ -0,0 +1,164 @@
+            type: item.tegund,
+            nationalId: item.kennitala,
+            name: item.nafn,
+          } as RskCompanyRelatedParty
+        }),
+        vat: company.virdisaukaskattur?.map((item) => {
+          return {
+            vatNumber: item.vskNumer,
+            dateOfRegistration: new Date(item.skrad),
+            status: item.stada,
+            dateOfDeregistration: new Date(item.afskraning),
+            classification: item.flokkun?.map(
+              (classification) =>
+                ({
+                  type: classification.gerd,
+                  classificationSystem: classification.flokkunarkerfi,
+                  number: classification.numer,
+                  name: classification.heiti,
+                } as RskCompanyClassification),
+            ),
+          } as RskCompanyVat
+        }),
+      },
+      lastUpdated: company.sidastUppfaert
+        ? new Date(company.sidastUppfaert)
+        : undefined,
+    }
+  }
+
+  async companyInformationSearch(
+    searchTerm: string,
+    first: number,
+    after = '',
+  ): Promise<RskCompanySearchItems | null> {
+    const searchResults = await this.rskCompanyInfoApi.v1FyrirtaekjaskraSearchSearchStringGet(
+      { searchString: searchTerm },
+    )
+    if (
+      !searchResults.items ||
+      !searchResults?.count ||
+      searchResults.count < 1
+    ) {
+      throw new NotFoundException('No search results found')
+    }
+
+    const formattedSearchResults = searchResults.items?.map((item) => {
+      return {
+        name: item.nafn,
+        status: item.stada,
+        dateOfRegistration: item.skrad ? new Date(item.skrad) : undefined,
+        nationalId: item.kennitala,
+        vatNumber: item.vskNumer,
+        lastUpdated: item.sidastUppfaert
+          ? new Date(item.sidastUppfaert)
+          : undefined,
+      } as RskCompany
+    })
+
+    const pageResult = this.getPage(formattedSearchResults, first, after)
+    if (!pageResult) {
+      throw new NotFoundException('No search results found')
+    }
+
+    const totalLength = searchResults.count
+    const cursor = toCursorHash(this.getLastItemIdentifer(pageResult.items))
+
+    return {
+      items: pageResult.items,
+      count: totalLength,
+      pageInfo: {
+        endCursor: cursor,
+        hasNextPage: pageResult.hasNextPage,
+      },
+    }
+  }
+
+  getPage(
+    searchResult: RskCompany[],
+    first: number,
+    cursor: string,
+  ): PageResult {
+    if (searchResult) {
+      if (cursor === '')
+        return {
+          items: searchResult.slice(0, first),
+          hasNextPage: first < searchResult.length,
+        }
+
+      const index = searchResult.findIndex(
+        (x) => x.nationalId === fromCursorHash(cursor),
+      )
+      const start = index + 1
+
+      return {
+        items: searchResult.slice(start, start + first),
+        hasNextPage: start + first < searchResult.length,
+      }
+    }
+    return { items: [], hasNextPage: false }
+  }
```

Comment:
This is only doing pagination in our graphql layer.

Does the original API not have any pagination setup? I assumed it used "links" for this.

Reply:
Great, let's pipe pagination directly to the backend. The cursor can be the offset for the next page, maybe base64 encoded. cc @saevarma